backend/src/test/java/codezap/global/DatabaseCleaner.java

```diff
@@ -29,4 +30,25 @@ private void truncate(List<String> queries, JdbcTemplate jdbcTemplate) {
         queries.forEach(jdbcTemplate::execute);
         jdbcTemplate.execute("SET FOREIGN_KEY_CHECKS = 1;");
     }
+
+    private void recreateFullTextIndex(JdbcTemplate jdbcTemplate) {
+        dropIndexIfExists(jdbcTemplate, "template", "idx_template_fulltext");
+        dropIndexIfExists(jdbcTemplate, "source_code", "idx_source_code_fulltext");
+
+        jdbcTemplate.execute("ALTER TABLE template ADD FULLTEXT INDEX idx_template_fulltext (title, description)");
+        jdbcTemplate.execute("ALTER TABLE source_code ADD FULLTEXT INDEX idx_source_code_fulltext (content, filename)");
+    }
+
+    private void dropIndexIfExists(JdbcTemplate jdbcTemplate, String tableName, String indexName) {
+        String checkIndexQuery = "SELECT COUNT(*) FROM information_schema.statistics " +
+                "WHERE table_schema = DATABASE() " +
+                "AND table_name = ? " +
+                "AND index_name = ?";
+
+        int indexCount = jdbcTemplate.queryForObject(checkIndexQuery, Integer.class, tableName, indexName);
+
+        if (indexCount > 0) {
+            jdbcTemplate.execute("ALTER TABLE " + tableName + " DROP INDEX " + indexName);
+        }
```

Comment:
있을 경우 drop하고 생성하는 로직인 것 같아요.
DB 접근을 줄이기 위해 있을 경우 놔두고 없을 경우 create해주는 건 어떨까요?
인자로 인덱스 주고 싶은 필드를 받으면 가능할 것 같아요 ~!

```suggestion
        if (indexCount == 0) {
            jdbcTemplate.execute("ALTER TABLE " + tableName + " ADD FULLTEXT INDEX " + indexName + " (" + String.join(", ", fields) + ")");
        }
```

Reply:
켬미 굿~! 

테스트 데이터가 많지는 않아서 DROP해도 괜찮을 거라고 생각하고 구현하긴 했는데 좋은 피드백이네요 👍

`CREATE FULLTEXT INDEX idx_source_code_fulltext ON source_code (content, filename);`
ALTER 문 말고 CREATE 를 사용하면 이미 있는 경우에 에러 메시지는 발생하지만 실행은 무시하더라구요?!

CREATE을 사용해서 DROP 하지 않도록 변경했습니다!